berth_reservations/logging.py

```diff
@@ -0,0 +1,52 @@
+import logging
+import time
+
+logger = logging.getLogger("requests")
+
+
+__all__ = ["LogRequestFilter", "RequestLogger"]
+
+
+class LogRequestFilter(logging.Filter):
+    """Since the requests are being logged by the RequestLogger,
+    we filter the django.server logs to avoid having duplicate logs"""
+
+    def filter(self, record: logging.LogRecord):
+        return record.name != "django.server"
+
+
+class RequestLogger:
```

Comment:
Other middleware seem to inherit from `MiddlewareMixin`, which adds async capability to the call. Might be useful here as well?

Reply:
The `MiddlewareMixin` is to add backwards compatibility to the [pre-Django 1.10 middlewares](https://docs.djangoproject.com/en/3.1/topics/http/middleware/#upgrading-pre-django-1-10-style-middleware).

The [example](https://docs.djangoproject.com/en/3.1/topics/http/middleware/#writing-your-own-middleware) on the django docs doesn't add it, and since we're not using the previous middleware style, we don't need it.
```python
class SimpleMiddleware:
    def __init__(self, get_response):
        self.get_response = get_response
        # One-time configuration and initialization.

    def __call__(self, request):
        # Code to be executed for each request before
        # the view (and later middleware) are called.

        response = self.get_response(request)

        # Code to be executed for each request/response after
        # the view is called.

        return response
```

Reply:
Oh, I see. The django >=3.1 way to add async support seems to be with the decorators, e.g. `@sync_and_async_middleware`. But I don't think there's much use to add that complexity here.

Reply:
Yes! We're not using async for views, so not really necessary here. 👍